### PR TITLE
Fix NPE causing stuck queries in insert returning

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

If there is a failure on an individual item the results are `null` which 
caused a NullPointerException in the result collection for `RETURNING`.

An example where the failure happened is if the insert contains a duplicate
primary key.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/test/java/io/crate/integrationtests/PostgresITest.java [hunk 1]
```diff
@@ -905,6 +907,38 @@ public void test_each_non_array_pg_type_entry_can_be_joined_with_pg_proc() throw
...
+    @Test
+    public void test_insert_from_unnest_returning_which_inserts_duplicate_keys() throws Exception {
+        try (var conn = DriverManager.getConnection(url(RW))) {
+            var statement = conn.createStatement();
+            statement.execute("create table tbl (id bigint primary key, val text)");
+            var textGenerator = DataTypeTesting.getDataGenerator(DataTypes.STRING);
+            var insertUnnest = conn.prepareStatement(
+                "insert into tbl (id, val) (select id, val from unnest(?, ?) as t (id, val)) returning _id");
+
+            for (int iteration = 0; iteration < randomIntBetween(2, 7); iteration++) {
```
Comment (seut) [inline server/src/test/java/io/crate/integrationtests/PostgresITest.java:919]: do we need more than 2 iterations? couldn't this be tested by adding a duplicate id inside the `ids` array? already

Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/PostgresITest.java:919]: I was thinking of the threshold where jdbc switched to binary streaming to have that covered too - but it's actually already randomized in the `PostgresITest` setup, so I can fix it to 2 iterations here :+1:

Comment (mfussenegger) [reply server/src/test/java/io/crate/integrationtests/PostgresITest.java]: @seut pushed a fixup. I tried to change it to have duplicate ids in just one request, but then the error doesn't occur. I think it only happens if there is no result rows at all.

**Review by seut — APPROVED** [review]
👍
